packages/frontend/src/Build.svelte

```diff
@@ -67,53 +66,80 @@ async function fillBuildOptions() {
   }
 }
 
-async function buildBootcImage() {
-  // Before ANYTHING validate / check all values before submitting:
-  // Reset the form validation error message
-  errorFormValidation = '';
+async function validate() {
+  let prereqs = await bootcClient.checkPrereqs();
+  if (prereqs) {
+    errorFormValidation = prereqs;
+    existingBuild = false;
+    return;
+  }
+
+  if (!selectedImage) {
+    errorFormValidation = 'No image selected';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildFolder) {
+    errorFormValidation = 'No output folder selected';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildType || buildType.length === 0) {
+    errorFormValidation = 'Must select at least one disk image type';
+    existingBuild = false;
+    return;
+  }
+
+  if (!buildArch) {
+    errorFormValidation = 'At least one architecture must be selected';
+    existingBuild = false;
+    return;
+  }
+
+  // overwrite
+  existingBuild = await bootcClient.buildExists(buildFolder, buildType);
+  if (existingBuild && !overwrite) {
+    errorFormValidation = 'Confirm overwriting existing build';
+    return;
+  }
+
+  // no problems, ready to build!
+  errorFormValidation = undefined;
+}
 
-  // Fields to check and it's output
-  const fieldsToCheck = [
-    { key: 'image name', value: buildImageName },
-    { key: 'tag', value: buildTag },
-    { key: 'engine ID', value: buildEngineId },
-    { key: 'build folder', value: buildFolder },
-    { key: 'build type', value: buildType },
-    { key: 'architecture', value: buildArch },
-  ];
-
-  // Go through and check if any of the fields are missing
-  // and create an error message if so
-  let missingFields: string[] = [];
-  fieldsToCheck.forEach(field => {
-    if (!field.value) missingFields.push(field.key);
-  });
-  if (missingFields.length > 0) {
-    errorFormValidation = `Missing fields: ${missingFields.join(', ')}`;
+async function buildBootcImage() {
+  // you can't get here without a selected image, but this
+  // avoids a svelte error
+  if (!selectedImage) {
     return;
   }
 
   // Before building a disk image name, we get a unique unused identifier for this image
   // This is to prevent the user from accidentally overwriting an history
+  const buildImageName = selectedImage.split(':')[0];
   const buildID = await bootcClient.generateUniqueBuildID(buildImageName);
 
   // The build options
+  const image = findImage(selectedImage);
   const buildOptions: BootcBuildInfo = {
     id: buildID,
     image: buildImageName,
-    tag: buildTag,
-    engineId: buildEngineId,
+    tag: selectedImage.split(':')[1],
+    engineId: image?.engineId ?? '',
```

Comment:
don't we always have to have an engine id for the bootc iamge build info?

Reply:
Same as above, this is what it was doing before. I had to add the ? and ?? to make the compiler happy and provide a fallback, but there aren't any cases where the image wouldn't be found.